module/strategy/WalkInsideBoundedBox/src/WalkInsideBoundedBox.cpp

```diff
@@ -0,0 +1,96 @@
+
+            log<NUClear::DEBUG>("cfg.defending_region ", cfg.defending_region.transpose());
+        });
+
+        on<Provide<DefendTask>, Trigger<Ball>, With<Field>, With<Sensors>>().then(
+            [this](const DefendTask& defend_task, const Ball& ball, const Field& field, const Sensors& sensor) {
+                Eigen::Isometry3f Hfw = Eigen::Isometry3f(field.Hfw.cast<float>());
+                Eigen::Isometry3f Hrw = Eigen::Isometry3f(sensor.Hrw.cast<float>());
+                Eigen::Isometry3f Hfr = Hfw * Hrw.inverse();
+                Eigen::Vector3f rBFf  = Hfw * ball.rBWw;
+                Eigen::Vector3f rRFf  = Hfr.translation();
+
+                // The distance of the robot from the ball
+                double robot_distance_to_ball =
+                    sqrt(abs(pow(rBFf.x() - rRFf.x(), 2.0) - pow(rBFf.y() - rRFf.y(), 2.0)));
+
+                log<NUClear::DEBUG>("rRFf: ", rRFf.transpose());
+                log<NUClear::DEBUG>("rBFf: ", rBFf.transpose());
+                log<NUClear::DEBUG>("Dist: ", robot_distance_to_ball);
+
+                // Check if the ball is in the defending region
+                if (rBFf.x() > cfg.defending_region(0) && rBFf.x() < cfg.defending_region(1)
+                    && rBFf.y() > cfg.defending_region(2) && rBFf.y() < cfg.defending_region(3)) {
+
+                    log<NUClear::DEBUG>("Ball inside of defending region");
+
+                    // Do nothing, play normally
+                }
+                else {
+                    // Robot - Defending position on field
+                    Eigen::Vector3f rDFf = Eigen::Vector3f::Zero();
+
+                    log<NUClear::DEBUG>("rDFf x: ", rDFf.x());
+                    log<NUClear::DEBUG>("rDFf y: ", rDFf.y());
+
+                    // If ball is in own half and outside the defending bounding box of robot
+                    if (rBFf.x() >= 0 && rBFf.y() > cfg.defending_region(3)) {
+                        log<NUClear::DEBUG>("Ball is in own half and in other region");
+
+                        // Calculate the defender position
+                        // Clamps to x direction of the ball and bounding box an 1 metre behind the ball
+                        rDFf.x() = std::clamp(rBFf.x(), cfg.defending_region(0), cfg.defending_region(1));
+                        rDFf.y() = std::clamp(rBFf.y(), cfg.defending_region(2), cfg.defending_region(3));
+                        rDFf.x() += 1.0;
+                    }
+                    else {
+                        // Calculate the defender position
+                        // Robot clamped to defending bounding box
+                        rDFf.x() = std::clamp(rBFf.x(), cfg.defending_region(0), cfg.defending_region(1));
+                        rDFf.y() = std::clamp(rBFf.y(), cfg.defending_region(2), cfg.defending_region(3));
+                    }
+
+                    // Walk to determined position
```

Comment:
what is this determined position? can you give more insight to what it might be?